flymake-vale.el

```diff
@@ -139,6 +139,34 @@ Passing the results and source BUF to CALLBACK."
   (or  (string-equal event "finished\n")
        (string-match "exited abnormally with code 1.*" event)))
 
+(defun flymake-vale--guess-extension-from-mode (mode)
+  "This is not good enough. At all.
+
+I don't know of a good way of getting the likely file extension of a
+major mode. The closest I found was the `auto-mode-alist' var, which
+has a regex value for all the file extensions which might match.
+
+Ideally, there'd be a way to generate a valid extension from the
+regexp, or there'd be a way of deciding a default suggested extension
+for a major-mode, but instead of wading in that right now and to prove
+the concept instead, I used this hack."
+  (let ((match (rassoc mode auto-mode-alist)))
+    (cond ((null match) ".txt")
+          (t (string-replace "?" "" (string-replace "'" "" (string-replace "\\" "" (car match))))))))
```

Comment:
I'd suggest either passing a successful value here or nil. Since a user can have specific rules for `txt` files, I do not think we should have it be the default.